### ISSUE TITLE
Jetpack: Error report 7.2.1: IllegalArgumentException: All roles must be handled

### ISSUE THREAD

**Issue by tonyr59h** [wordpress-mobile/WordPress-Android#5852]
Looks like we're there's a "Pending" role that the app is unaware of.

```
Non-fatal Exception: java.lang.IllegalArgumentException: All roles must be handled: pending
       at org.wordpress.android.models.Role.fromString(Role.java:53)
       at org.wordpress.android.models.Person.userFromJSON(Person.java:50)
       at org.wordpress.android.ui.people.utils.PeopleUtils.peopleListFromJSON(PeopleUtils.java:303)
       at org.wordpress.android.ui.people.utils.PeopleUtils.access$000(PeopleUtils.java:24)
       at org.wordpress.android.ui.people.utils.PeopleUtils$1.onResponse(PeopleUtils.java:36)
       at org.wordpress.android.ui.people.utils.PeopleUtils$1.onResponse(PeopleUtils.java:30)
       at com.wordpress.rest.RestRequest.deliverResponse(RestRequest.java:76)
       at com.wordpress.rest.RestRequest.deliverResponse(RestRequest.java:19)
       at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:99)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5417)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)```

**Comment by tonyr59h** [comment]
The crash logs contain various undefined roles, such as `pending`, `gamer`, and `customer`. The solution will have to handle generic roles more gracefully.

**Comment by tonyr59h** [comment]
Some notes:
* This is a Jetpack issue, signing into a Jetpack account with sites that have custom roles is not handled by the app. Users with unrecognized roles end up defaulting to `Contributor`.
* Users with no assigned role are also considered `Contributor`
* Custom roles are not available when assigning roles or adding new users
* REST endpoint (v1.1) for listing roles is not documented but available and working (`/sites/$site/roles`)
* Localization via the endpoint does not seem to work (passing `?lang=es` to test)

I propose we add a `CUSTOM` case to the `Role` enum to handle non-standard roles. Then we can remove this Crashlytics exception.

**Comment by maxme** [comment]
Added the Jetpack label, we can work on this during our jetpack sprint.

**Comment by tonyr59h** [comment]
Cool, I'll remove myself for the time being in that case.

**Comment by oguzkocer** [comment]
Jetpack sites should not be able to browse the People page for now. That sounds like a bug to me. I'll check that today and disable it if it's the case. This is one of the many issues we'll have to resolve about the People Management for Jetpack sites.

**Comment by maxme** [comment]
> Jetpack sites should not be able to browse the People page for now

They should: https://github.com/wordpress-mobile/WordPress-Android/issues/5123

And I think we should keep this, it works well for the majority of Jetpack users. And it's OK to have a limited experience for Jetpack power-users in the meantime.

**Comment by maxme** [comment]
Just noticed it's a crash, we should add a temporary fix for these users until we support any kind of roles.

**Comment by tonyr59h** [comment]
> Just noticed it's a crash

I don't think this is a crash, just us logging unhandled state via `Crashlytics.logException`. It ends up defaulting to `Role.CONTRIBUTOR`; relevant comment at the end of `fromString`:

```
// All roles should have been handled, but in case an edge case occurs,
// using "Contributor" role is the safest option```

**Comment by tonyr59h** [comment]
Closed via #6397.
